src/main/java/com/codersbay/part2/RingBuffer.java

```diff
@@ -61,7 +61,21 @@ public boolean isFull() {
      * @param item to be appended to the buffer.
      */
     public void enqueue(String item) {
+        if (numberOfElementsOnQueue < queue.length) {
+            numberOfElementsOnQueue++;
+        }else {
+            if (first == queue.length) {
+                first =0;
+            }else {
+                first++;
+            }
+        }
+        last++;
 
+        if (last == queue.length) {
+            last = 0;
+        }
+        queue[last] = item;
```

Comment:
Not correct `testEnqueue` test fails, can be done in:

~~~java
if (size() == capacity()) {
	queue[0] = item;

} else {
	numberOfElementsOnQueue++;
	queue[last] = item;
}
~~~